docs/ref/settings.txt

```diff
@@ -2764,11 +2764,12 @@ See also :setting:`LANGUAGE_CODE`, :setting:`USE_I18N` and :setting:`USE_TZ`.
 Default: ``False``
 
 A boolean that specifies whether to display numbers using a thousand separator.
-When :setting:`USE_L10N` is set to ``True`` and if this is also set to
-``True``, Django will use the values of :setting:`THOUSAND_SEPARATOR` and
-:setting:`NUMBER_GROUPING` to format numbers unless the locale already has an
-existing thousands separator. If there is a thousands separator in the locale
-format, it will have higher precedence and will be applied instead.
+When set to ``True`` and :setting:`USE_L10N` is also ``True``, Django will
+format numbers using the values of the :setting:`NUMBER_GROUPING` setting and
+the locale's ``THOUSAND_SEPARATOR`` format (if defined) or the
```

Comment:
I think it's still a bit misleading, because it suggests that we always use the  `NUMBER_GROUPING` settings, which is not true. The locale-dictated format has higher precedence also for it.

Reply:
See ticket-25928, and 
https://github.com/django/django/blob/5ef6f626347f9ea13915f3cf5b8b045c5b42b102/django/utils/formats.py#L176-L184

https://github.com/django/django/blob/5ef6f626347f9ea13915f3cf5b8b045c5b42b102/django/utils/numberformat.py#L21

https://github.com/django/django/blob/5ef6f626347f9ea13915f3cf5b8b045c5b42b102/django/utils/numberformat.py#L70-L89

Reply:
Right - I was missing that description. I've tried a new wording, what do you think?

Reply:
Sound good :+1: